crates/agent/src/main.rs

```diff
@@ -210,13 +210,31 @@ async fn async_main(args: Args) -> Result<(), anyhow::Error> {
         id_gen.clone(),
         connectors,
     );
+
+    let mut data_planes =
+        agent_sql::data_plane::fetch_data_planes(&pg_pool, Vec::new(), "", uuid::Uuid::nil())
+            .await?;
+
+    futures::future::try_join_all(
+        data_planes
+            .iter_mut()
+            .map(|dp| agent::decrypt_hmac_keys(dp)),
+    )
+    .await?;
+
+    let decrypted_hmac_keys = data_planes
+        .iter()
+        .map(|dp| (dp.data_plane_name.clone(), dp.hmac_keys.clone()))
+        .collect();
+
     let control_plane = agent::PGControlPlane::new(
         pg_pool.clone(),
         system_user_id,
         publisher.clone(),
         id_gen.clone(),
         discover_handler.clone(),
         logs_tx.clone(),
+        decrypted_hmac_keys,
```

Comment:
Turns out it is somewhat more tricky to get this working the same way as it works in Snapshot (i.e. refresh the decrypted hmac_keys when data planes are updated, etc.) since it requires a mutable reference to ControlPlane, which is fine until we hit [here](https://github.com/estuary/flow/blob/master/crates/agent/src/controllers/executor.rs#L30), and we could have `Arc<Mutex<_>>` here but I didn't want to complicate things further by introducing a lock here. I think that would mean LiveSpecControllerExecutors can no longer run in parallel due to the lock...

I'm trying to find a way to refresh the decrypted hmac keys that doesn't lead to a lock 🤔

Reply:
Does it get easier at all to have it re-use `agent::api::App` ? Definitely a Frankenstein, but direction-ally we do want to move them closer together I think...

Reply:
@jgraettinger I implemented a similar cache refreshing mechanism as the `fetch_loop` of Snapshots here

This uses an `RwLock` which distinguishes between reads and writes, so multiple controllers reading the same value won't lock it, but during the write, reads will be blocked (every minute). I think this is a reasonable approach for now